### PR TITLE
[IOS-38] Refactor Retain Cycle issues & Order of Operations in HyBid SDK

### PR DESCRIPTION
**Pull request by can-soykarafakili**

### CODE REVIEW

File: PubnativeLite/PubnativeLite/Ad Presenter/PNLiteAdPresenterDecorator.m [hunk 1]
```diff
@@ -65,17 +65,24 @@ - (instancetype)initWithAdPresenter:(HyBidAdPresenter *)adPresenter
 #pragma mark HyBidAdPresenterDelegate
 
 - (void)adPresenter:(HyBidAdPresenter *)adPresenter didLoadWithAd:(UIView *)adView {
-    [self.adTracker trackImpression];
-    [self.adPresenterDelegate adPresenter:adPresenter didLoadWithAd:adView];
+    
+    if (self.adPresenterDelegate && [self.adPresenterDelegate respondsToSelector:@selector(adPresenter:didLoadWithAd:)]) {
```
Comment (eros902002) [inline PubnativeLite/PubnativeLite/Ad Presenter/PNLiteAdPresenterDecorator.m]: In which situation would the presenter not have the **didLoadWithAd** function? Same for all the ones down.

File: PubnativeLite/PubnativeLite/Consent/HyBidUserDataManager.m [hunk 2]
```diff
@@ -70,7 +70,7 @@ + (instancetype)sharedInstance {
     return sharedInstance;
 }
 
-- (void)createUserDataManagerWithAppToken:(NSString *)appToken completion:(UserDataManagerCompletionBlock)completion {
```
Comment (eros902002) [inline PubnativeLite/PubnativeLite/Consent/HyBidUserDataManager.m]: Why do we remove this param? Was it not used?

File: PubnativeLite/PubnativeLite/Ad Presenter/PNLiteAdPresenterDecorator.m (hunk 1 above)
Comment (can-soykarafakili) [reply PubnativeLite/PubnativeLite/Ad Presenter/PNLiteAdPresenterDecorator.m]: It is mainly checking for the delegate is whether nil or not.

File: PubnativeLite/PubnativeLite/Consent/HyBidUserDataManager.m (hunk 2 above)
Comment (can-soykarafakili) [reply PubnativeLite/PubnativeLite/Consent/HyBidUserDataManager.m]: We are using the appToken that is saved to the HyBid sharedInstance

**Pull request merged**